wordpress-mobile/WordPress-Android
# Location handling cleanup

This WIP PR takes care of "cleaning up" after switching to per-app language preferences. We had a lot of code to handle locale changes, but the system now handles all of that for us. Testing steps are the same as with previous PRs:

Test 1:

- Run the app on a device using Android 13 or later
- Go to App Settings and tap "Interface language"
- Note that the system dialog for "App Language" appears instead of our own
- Verify changing languages works

Test 2:

- Perform the above with an older device and verify our own language dialog appears and changing languages works

**Note:** There is likely more cleaning up that can be done, but I didn't want to make this PR even larger that it already is.
